src/client/src/Client.c

```diff
@@ -94,6 +94,57 @@ STATUS checkIntermittentProducerCallback(UINT32 timerId, UINT64 currentTime, UIN
     return retStatus;
 }
 
+STATUS configureClientWithRetryStrategy(PKinesisVideoClient pKinesisVideoClient, KVS_RETRY_STRATEGY_TYPE kvsRetryStrategyType) {

```

Comment:
The declaration of this fn needs to go into the internal header file.

Most importantly, design of the API... You can take a moniker of the action (an enum defining what to do) or the action itself. The first is more compact and easy to use for a client interface but is very limited and hard to expand. The latter, providing the actual functions, is more flexible. Taking that this is an internal function, you could go with both approaches. I prefer the latter for testing purposes where the test could control the setting of the functions in mid-flight for example.

Reply:
This is actually a private function. In the newer commit, I've added a client callback to define retry strategy. With the existing approach, it is nearly impossible to unit test this code. So I had to add a callback in client. Please check the newer code. That might answer your concern.